include/interfaces/scalar_reaction_diffusion_convection_stabilized.h

```diff
@@ -0,0 +1,224 @@
+/*! \addtogroup equations
+ *  @{
+ */
+
+/**
+ *  This interface solves a Scalar Reaction Diffusion Convection Equation:
+ *  \f[ TODO:
+ *       u_t - \nu * \Delta u + b\cdot\nabla u + cu = f
+ *  \f]
+ */
+
+#ifndef _pidoums_scalar_reaction_diffusion_convection_h_
+#define _pidoums_scalar_reaction_diffusion_convection_h_
+
+#include "pde_system_interface.h"
+
+#include <deal.II/lac/solver_gmres.h>
+
+#include <deal2lkit/parsed_function.h>
+#include <deal2lkit/parsed_preconditioner/amg.h>
+
+//typedef LATrilinos LAC;
+
+template <int dim, int spacedim, typename LAC=LATrilinos>
+class ScalarReactionDiffusionConvection : public PDESystemInterface<dim,spacedim, ScalarReactionDiffusionConvection<dim,spacedim,LAC>, LAC>
+{
+
+public:
+  ~ScalarReactionDiffusionConvection () {};
+  ScalarReactionDiffusionConvection ();
+
+  void declare_parameters (ParameterHandler &prm);
+
+  template <typename EnergyType, typename ResidualType>
+  void energies_and_residuals(const typename DoFHandler<dim,spacedim>::active_cell_iterator &cell,
+                              FEValuesCache<dim,spacedim> &scratch,
+                              std::vector<EnergyType> &energies,
+                              std::vector<std::vector<ResidualType> > &local_residuals,
+                              bool compute_only_system_terms) const;
+
+
+  void compute_system_operators(const std::vector<shared_ptr<LATrilinos::BlockMatrix> >,
+                                LinearOperator<LATrilinos::VectorType> &,
+                                LinearOperator<LATrilinos::VectorType> &,
+                                LinearOperator<LATrilinos::VectorType> &) const;
+
+private:
+
+  double nu;
+
+  /**
+   * Stabilization parameter
+   */
+  double alpha;
+
+  /**
+   * Reaction parameter
+   */
+  double gamma;
+
+  /**
+   * AMG preconditioner for velocity-velocity matrix.
+   */
+  mutable ParsedAMGPreconditioner AMG_A;
+
+  /**
+   * AMG preconditioner for velocity-velocity matrix.
+   */
+  mutable ParsedAMGPreconditioner AMG_B;
+
+  ParsedFunction<dim> convection;
+};
+
+template <int dim, int spacedim, typename LAC>
+ScalarReactionDiffusionConvection<dim,spacedim, LAC>::
+ScalarReactionDiffusionConvection():
+  PDESystemInterface<dim,spacedim,ScalarReactionDiffusionConvection<dim,spacedim,LAC>, LAC >("Convection Diffusion Reaction Problem",
+      1 + dim,
+      1,
+      "FESystem[FE_Q(2)^d-FE_Q(1)]",
+      dim==2 ? "u,q,q" : "u,q,q,q",
```

Comment:
So, this is a Q2^d - Q1, but you name the variables u,q,q, instead of u,u,q (or q,q,u). Is this the intended behaviour?